### PR TITLE
Add `excludeInactiveDatasets` to `CkanCatalogGroup`

### LINKED ISSUE #1143: DGA CKAN filter out "data_state" = "inactive"
See https://github.com/TerriaJS/terriajs/pull/6255#pullrequestreview-949331003

This could be done using by implementing something like `excludeMembersByProperties` at the `GroupMixin` level.

Eg - http://ci.terria.io/main/#configUrl=https%3A%2F%2Fnationalmap.gov.au%2Fconfig.json&share=s-nxnnxNRNJKbbAPYNX2jzTID5k7M

image

### PR DESCRIPTION
**Pull request by nf-s**
### Add `excludeInactiveDatasets` to `CkanCatalogGroup`

Fixes https://github.com/TerriaJS/nationalmap/issues/1143

Added `excludeInactiveDatasets` to `CkanCatalogGroup` (`true` by default). This will filter out CKAN Datasets which have `state` or `data_state` (data.gov.au specific) **not** set to `"active"`.

### After vs Before

In NationalMap data.gov.au CKAN - lots of problematic organisations have been excluded

image

### Test me

- **Before** http://ci.terria.io/main/#configUrl=https://terria-catalogs-public.storage.googleapis.com/nationalmap/config/dev.json
- **After** http://ci.terria.io/ckan-filter-datasets/#configUrl=https://terria-catalogs-public.storage.googleapis.com/nationalmap/config/dev.json

### Checklist

- [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
- [ ] ~I've updated relevant documentation in `doc/`.~
- [x] I've updated CHANGES.md with what I changed.
- [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/Models/Catalog/Ckan/CkanCatalogGroup.ts [hunk 1]
```diff
@@ -261,6 +261,15 @@ export class CkanServerStratum extends LoadableStratum(CkanCatalogGroupTraits) {
       return;
     }
 
+    // Return if excluding inactive dataset and dataset is inactive
+    if (
```
Comment (staffordsmith83) [inline lib/Models/Catalog/Ckan/CkanCatalogGroup.ts:269]: Is it possible to have a case where only one of `state` or `data_state` is active and that means it's still a valid working datasource? Like state=active and dataState=inactive? Probably not just checking.

Also, is the only allowable value for `dataState` or `state` "active"? Is this a ckan standard? I see in the docs:

"state (string) – the current state of the dataset, e.g. 'active' or 'deleted', only active datasets show up in search results and other lists of datasets, this parameter will be ignored if you are not authorized to change the state of the dataset (optional, default: 'active')"

But it does say "e.g. 'active' or 'deleted'" implying that other values can be used here... Is it possible we need to account for a wider range of values? Current logic just says that if the flag is present, and it has any other value than active, dont show it... Reckon thats OK?

File: lib/Traits/TraitsClasses/CkanCatalogGroupTraits.ts [hunk 2]
```diff
@@ -60,4 +60,11 @@ export default class CkanCatalogGroupTraits extends mixTraits(
     description: `True to allow entire WMS servers (that is, WMS resources without a clearly-defined layer) to be added to the catalog; otherwise, false.`
   })
   allowEntireWmsServers: boolean = true;
+
+  @primitiveTrait({
+    type: "boolean",
+    name: "Exclude inactive datasets",
+    description: `True to remove inactive datasets (\`state\` is not \`active\` OR )\`data_state\` is not \`active\`.`
```
Comment (staffordsmith83) [inline lib/Traits/TraitsClasses/CkanCatalogGroupTraits.ts]: How come the backticks around the entire description? Should I be doing this?

**Review by staffordsmith83 — CHANGES_REQUESTED** [review]
Just to make sure you have considered the other possible values for `state` - I dont know much about ckan so happy for a quick response and Ill approve.

I can see how its cleaned up heaps good idea!

File: lib/Traits/TraitsClasses/CkanCatalogGroupTraits.ts (hunk 2 above)
Comment (nf-s) [reply lib/Traits/TraitsClasses/CkanCatalogGroupTraits.ts]: Nah I don't know why I did that. I have changed to double quotes.

Comment (nf-s) [reply lib/Traits/TraitsClasses/CkanCatalogGroupTraits.ts]: Sometimes I prefer backticks if I have nested quotes and things in the description - as the field is rendered as markdown in our docs

File: lib/Models/Catalog/Ckan/CkanCatalogGroup.ts (hunk 1 above)
Comment (nf-s) [reply lib/Models/Catalog/Ckan/CkanCatalogGroup.ts:269]: From flicking through [ckan](https://github.com/ckan/ckan) code base - I can see the following values for `state`
- "active" 
- "deleted" 
- "draft"

As for `data_state` - we have the schema here https://github.com/datagovau/ckanext-datagovau/blob/966e045f8bfe0cc1453d9eaf6511e8c476d59c26/ckanext/datagovau/schemas/dataset.yaml#L91-L100

- "active" 
- "inactive" 

As for why data.gov.au has an additional property - I have no idea. All of the datasets we harvest through data.gov.au have `state = "active"`.

I have added some comments to clarify both properties. 

https://github.com/TerriaJS/terriajs/blob/9f9b414d1f7149eadcdbe39f8842378404a341de/lib/Models/Catalog/Ckan/CkanDefinitions.ts#L55-L62

I have changed logic to check for values other than `"active"`:

https://github.com/TerriaJS/terriajs/blob/0f6700a1eda14ea293a070b589cc65f35abe8e3d/lib/Models/Catalog/Ckan/CkanCatalogGroup.ts#L264-L276

Comment (staffordsmith83) [reply lib/Models/Catalog/Ckan/CkanCatalogGroup.ts:269]: Oh yeah that looks solid!

**Review by staffordsmith83 — APPROVED** [review]
Looks good man!

**Pull request merged**